PythonAPI/pycocotools/cocoeval.py

```diff
@@ -496,6 +528,161 @@ def _summarizeKps():
     def __str__(self):
         self.summarize()
 
+    def filter_annotations(self, annotations, catIds):
+        if hasattr(catIds, '__iter__') or hasattr(catIds, '__len__'):
+            catIds = catIds
+        else:
+            catIds = [catIds]
+
+        new_annotations = []
+        for anno in annotations:
+            if anno['category_id'] in catIds:
+                new_annotations.append(anno)
+
+        return new_annotations
+
+    def makeplot(self, recThrs, precisions, name, save_dir=None):
+        if save_dir is not None and not os.path.exists(save_dir):
+            os.makedirs(save_dir)
+        #precisions  7x101x4x1]
+        assert precisions.shape[2] == 4
+        areaNms = ['all','small','medium','large']
+        figures_np = {}
+        for aidx in range(precisions.shape[2]):
+            plotname = '%s-%s' %(name, areaNms[aidx])
+            #ps shape 7x101
+            ps = precisions[:, :, aidx, 0].squeeze()
+            ap = np.round(np.mean(ps, axis=1)*1000)
+
+            ds = np.concatenate((np.expand_dims(ps[0], axis=0), np.diff(ps, axis=0)))
+            colors = [
+                [1.0, 1.0, 1.0],
+                [0.5, 0.5, 0.5],
+                [.31, .51, .74],
+                [.75, .31, .30],
+                [.36, .90, .38],
+                [.50, .39, .64],
+                [1.0, .6,  0.0]
+            ]
+            # format legend str
+            ls = ['C75','C50','Loc','Sim','Oth','BG','FN']
+            for i in range(precisions.shape[0]):
+                if ap[i] == 1000:
+                    ls[i] = '[1.00] ' + ls[i]
+                else:
+                    ls[i] = '[.%03d] %s' %(ap[i], ls[i])
+
+            fig = plt.figure()
+            plt.stackplot(recThrs, ds, labels=ls, colors=colors)
+            plt.axis([0.0, 1.0, 0.0, 1.0])
+            plt.legend(loc='lower left')
+            plt.xlabel('recall')
+            plt.ylabel('precision')
+            plt.title(plotname)
+
+            # if save_dir is set, figure will be saved to save_dir
+            if save_dir is not None:
+                save_file = '%s/%s.pdf' % (save_dir, plotname)
+                print('save figure to %s' % save_file)
+                fig.savefig(save_file)
+            fig.canvas.draw()
+            data = np.fromstring(fig.canvas.tostring_rgb(), dtype=np.uint8, sep='')
+            data = data.reshape(fig.canvas.get_width_height()[::-1] + (3,))
+            figures_np[name] = data
+            plt.close()
+
+        return figures_np
+
+
+    def analyze(self):
```

Comment:
Would be great if we could set `outDir` from here!

Request for it to be a parameter:

```
def analyse(self, save_to_dir):
    ...
    self.params.outDir = save_to_dir
    ...
```

Reply:
done

Reply:
My apologies, I didn't fully understand the code at the time I suggested this. Doing this introduces a bug where the `overall` plots do not get saved